### ISSUE TITLE
ActiveJob does not handle Parameters serialization in rails5

### ISSUE THREAD

**Issue by swangs** [rails/rails#33643]
In Rails 5 ActionController:Parameters is not inheriting from HashWithIndifferentAccess. ActiveJob serialization raises an argument error when when trying to pass in a Parameters object. 

```ruby
      def serialize_argument(argument)
        case argument
        when *TYPE_WHITELIST
          argument
        when GlobalID::Identification
          convert_to_global_id_hash(argument)
        when Array
          argument.map { |arg| serialize_argument(arg) }
        when ActiveSupport::HashWithIndifferentAccess
          result = serialize_hash(argument)
          result[WITH_INDIFFERENT_ACCESS_KEY] = serialize_argument(true)
          result
        when Hash
          symbol_keys = argument.each_key.grep(Symbol).map(&:to_s)
          result = serialize_hash(argument)
          result[SYMBOL_KEYS_KEY] = symbol_keys
          result
        else
          Serializers.serialize(argument)
        end
      end
```
Maybe add in:
```ruby
        when ActionController::Parameters
          arguments = arguments.to_hash
          symbol_keys = argument.each_key.grep(Symbol).map(&:to_s)
          result = serialize_hash(argument)
          result[SYMBOL_KEYS_KEY] = symbol_keys
          result
```

**Comment by y-yagi** [comment]
I'm not sure if this is the guaranteed behavior.  
However, I think that `ActionController::Parameters` was not in the support type described in the guide, and it happened to be working by chance. 
https://edgeguides.rubyonrails.org/active_job_basics.html#supported-types-for-arguments

**Comment by rafaelfranca** [comment]
Fixed in 5-2-stable by 9d33e11893
